core/src/test/java/io/questdb/test/griffin/ExpressionParserTest.java

```diff
@@ -358,6 +358,69 @@ public void testCaseWithDanglingCast() {
         );
     }
 
+    @Test
+    public void testCountDistinctRewrites() throws SqlException {
+        x("foo count_distinct", "count_distinct(foo)");
+        x("foo count_distinct", "count(distinct foo)");
+
+        x("1 1 + count_distinct", "count_distinct(1+1)");
+        x("1 1 + count_distinct", "count(distinct 1+1)");
+
+        x("bar foo count_distinct", "count_distinct(foo(bar))");
+        x("bar foo count_distinct", "count(distinct foo(bar))");
+
+        x("bar foo count_distinct varchar cast", "cast (count_distinct(foo(bar)) as varchar)");
+        x("bar foo count_distinct varchar cast", "cast (count(distinct foo(bar)) as varchar)");
+
+        // 'distinct' on its own is treated as a column name. technically, it should have been in double quotes
+        // since it is a reserved word. but that's a compatibility issue. so let's check it's not rewritten
+        x("distinct count varchar cast", "cast (count(distinct) as varchar)");
+        // check it works even if quoted
+        x("distinct count varchar cast", "cast (count(\"distinct\") as varchar)");
+        // it works with multiple arguments too
+        x("distinct foo count", "count(\"distinct\", foo)");
+        // not written when count is not a function
+        x("count distinct foo", "foo(count, distinct)");
+        x("distinctnot count", "count(distinctnot)");
+        x("bar count distinct foo", "foo(bar, count, distinct)");
+        x("count bar distinct foo", "foo(bar(count), distinct)");
+
+        assertFail("count(distinct *)", 6, "count(distinct *) is not supported");
+        assertFail("count(distinct ", 6, "table and column names that are SQL keywords have to be enclosed in double quotes, such as \"distinct\"");
+        assertFail("notcount(distinct foo)", 18, "dangling literal");
+    }
+
+    @Test
+    public void testStringAggDistinctRewrites() throws SqlException {
```

Comment:
Postgres supports the following syntax for `string_agg`:  `string_agg(distinct col, ',' order by col asc)`. Would be nice to add tests around this syntax to make sure that we return a meaningful error message.